src/commonMain/kotlin/fr/acinq/lightning/wire/RouteBlinding.kt

```diff
@@ -10,6 +10,17 @@ import fr.acinq.bitcoin.utils.Either
 import fr.acinq.lightning.*
 
 sealed class RouteBlindingEncryptedDataTlv : Tlv {
+    /** Id of the next node. */
+    data class OutgoingPhoenixId(val nodeId: PublicKey) : RouteBlindingEncryptedDataTlv() {
+        override val tag: Long get() = OutgoingPhoenixId.tag
+        override fun write(out: Output) = LightningCodecs.writeBytes(nodeId.value, out)
+
+        companion object : TlvValueReader<OutgoingPhoenixId> {
+            const val tag: Long = 0
+            override fun read(input: Input): OutgoingPhoenixId = OutgoingPhoenixId(PublicKey(LightningCodecs.bytes(input, 33)))
+        }
+    }
```

Comment:
We can't use TLV type `0`, this is reserved for official TLVs from the BOLTs. If we introduce a separate TLV, we need to use a value greater than 65536, as detailed in [blip-0002](https://github.com/lightning/blips/blob/master/blip-0002.md#tlv-fields-in-bolt-messages).

But I think we can actually avoid introducing a new TLV here. Since you introduce this case as a new `EncodedNodeId`, why not do this consistently and just use the `OutgoingNodeId` TLV directly?

Right now there is a slightly weird asymmetry in `LightningCodecs.kt` where you handle the new `EncodedNodeId` when writing, but not when reading (and when you write it you lose the information that it's a phoenix `node_id`). We could instead keep this information when encoding:

- if the public key starts by `0x02`, replace that first byte by `0x04`
- if the public key starts by `0x03`, replace that first byte by `0x05` 
- when reading, it lets us detect that this is a phoenix `node_id` and transform it in the other direction (`0x04` -> `0x02` and `0x05` -> `0x03`)

Reply:
Very good idea.